internal/services/apimanagement/api_management_workspace_logger_resource.go

```diff
@@ -0,0 +1,435 @@
+		},
+
+		"eventhub": {
+			Type:         pluginsdk.TypeList,
+			MaxItems:     1,
+			Optional:     true,
+			ForceNew:     true,
+			ExactlyOneOf: []string{"application_insights", "eventhub"},
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"name": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validate.ValidateEventHubName(),
+					},
+
+					"connection_string": {
+						Type:      pluginsdk.TypeString,
+						Optional:  true,
+						Sensitive: true,
+						ExactlyOneOf: []string{
+							"eventhub.0.connection_string",
+							"eventhub.0.endpoint_uri",
+						},
+						ConflictsWith: []string{
+							"eventhub.0.user_assigned_identity_client_id",
+						},
+						ValidateFunc: validation.StringIsNotEmpty,
+					},
+
+					"endpoint_uri": {
+						Type:         pluginsdk.TypeString,
+						Optional:     true,
+						ValidateFunc: validation.StringIsNotEmpty,
+						ExactlyOneOf: []string{
+							"eventhub.0.connection_string",
+							"eventhub.0.endpoint_uri",
+						},
+					},
+
+					"user_assigned_identity_client_id": {
+						Type:         pluginsdk.TypeString,
+						Optional:     true,
+						ValidateFunc: validation.IsUUID,
+						ConflictsWith: []string{
+							"eventhub.0.connection_string",
+						},
+					},
+				},
+			},
+		},
+
+		"resource_id": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: azure.ValidateResourceID,
+		},
+	}
+}
+
+func (r ApiManagementWorkspaceLoggerResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (r ApiManagementWorkspaceLoggerResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.ApiManagement.LoggerClient_v2024_05_01
+
+			var model ApiManagementWorkspaceLoggerModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			workspaceId, err := logger.ParseWorkspaceID(model.ApiManagementWorkspaceId)
+			if err != nil {
+				return err
+			}
+
+			id := logger.NewWorkspaceLoggerID(workspaceId.SubscriptionId, workspaceId.ResourceGroupName, workspaceId.ServiceName, workspaceId.WorkspaceId, model.Name)
+			existing, err := client.WorkspaceLoggerGet(ctx, id)
+			if err != nil && !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for presence of existing %s: %+v", id, err)
+			}
+
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			parameters := logger.LoggerContract{
+				Properties: &logger.LoggerContractProperties{
+					IsBuffered: pointer.To(model.BufferingEnabled),
+				},
+			}
+
+			if len(model.ApplicationInsights) > 0 {
+				parameters.Properties.LoggerType = logger.LoggerTypeApplicationInsights
```

Comment:
It looks like there are 3 `LoogerTypes` at the moment. Managing all those in a single resource is not ideal and they should be split into their own resources